zero_bin/leader/src/client.rs

```diff
@@ -46,6 +46,20 @@ pub(crate) async fn client_main(
         ),
     ));
 
+    if let BlockInterval::FollowFrom { start_block, .. } = block_interval.clone() {
```

Comment:
I think you could add getter function to `BlockIInterval`:
```
pub fn get_start_block(&self) -> u64
```
instead of de-structuring here block interval.

Also, I believe checkpoint could be provided for all `BlockInterval` variants, not just for `FollowFrom` and `Range`.